### PR TITLE
Update informer.py

### PR DESCRIPTION
**Pull request by MarKovka20**
Added class LongTermAttention which actually make everything described in 3rd part of the article. The forward function takes keys (k) and queries (q) and return  long-term context representation Z_LTM and KL-divergence loss (kl_reg).

Further according to the article we need to add Z_LTM to regular transformer context representation Z_T and add kl_reg to regular nll_loss.

Any ideas how to implement it in current code (maybe in Baseline model)?

### CODE REVIEW

File: src/models/informer.py [hunk 1]
```diff
@@ -3,3 +3,439 @@
 
 class Informer(TorchModel, config_name='informer'):
     pass
+
+class GaussianBasisFunctions(object):
```
Comment (NonameUntitled) [inline src/models/informer.py:13]: We don't need t write object, we don't  use Python2
```suggestion
class GaussianBasisFunctions:
```

File: src/models/informer.py [hunk 2]
```diff
@@ -3,3 +3,439 @@
 
 class Informer(TorchModel, config_name='informer'):
     pass
+
```
Comment (NonameUntitled) [inline src/models/informer.py:11]: Formatting errors, you need two empty lines after class definition. Check PEP8 style-guide

File: src/models/informer.py [hunk 3]
```diff
@@ -3,3 +3,439 @@
...
+class GaussianBasisFunctions(object):
+    """Function phi(t) = Gaussian(t; mu, sigma_sq)."""
+    def __init__(self, mu, sigma):
+        self.mu = mu.unsqueeze(0)
+        self.sigma = sigma.unsqueeze(0)
+
+    def __repr__(self):
+        return f"GaussianBasisFunction(mu={self.mu}, sigma={self.sigma})"
```
Comment (NonameUntitled) [inline src/models/informer.py:20]: Let's use unit quotes be default
```suggestion
        return f'GaussianBasisFunction(mu={self.mu}, sigma={self.sigma})'
```

File: src/models/informer.py [hunk 4]
```diff
@@ -3,3 +3,439 @@
...
+            return context, kl_reg
+        else:
+            return context
+        
+    @property
+    def _query_dim(self):
+        return self.query_layer.in_features
+
+    def __repr__(self):
+        return "ContinuousAttention"
```
Comment (NonameUntitled) [inline src/models/informer.py]: No empty line at the EOF, please, add empty line here

**Comment by MarKovka20** [comment]
The Long Term Attention class has been redesigned relative to the initial implementation from the github of the article. The parameters that are not necessary for our task have been removed. The SimpleInformerModel class is written, which is similar to the BaselineModel class. It combines Baseline Projector and implemented Long Term Attention.

Some experiments comparing their baseline model and the resulting informer are given in the notebook informer_testing_vs_baseline.ipynb.

 (in short, the informer works faster, but it doesn't look like it trains, we need to check the LongTermAttention class, maybe play with hyperparameters and/or do as in the original article, in which long term attention was summed up with the usual one)

TODO: establish work with config files
TODO: change the train function so that we can use kl_loss

**Review by NonameUntitled — CHANGES_REQUESTED** [review]

**Pull request merged**